File: filesystem/lib/cd/command.lua
function entrypoint()
    local to = path
    if to == nil then
        to = "."
    end

    if to == ".." then
        workspace.dir = fs.parentDir(workspace.dir)
    else
        if fs.exists(to) then
            workspace.dir = to
        else
            error("Directory does not exist.")
        end
    end

    return fs.absolute(workspace.dir)
end

File: filesystem/lib/cd/command.py
import sys
from pathlib import Path

import shfl


def entrypoint(ws, options, bg_mode):
    currentDir = Path(shfl.currentDirectory(ws))
    if options["path"] == "..":
        shfl.moveDirectory(ws, str(currentDir.parent))
    else:
        to = currentDir / options["path"]
        if to.exists():
            shfl.moveDirectory(ws, str(to))
        else:
            print("Directory does not exist.", file=sys.stderr)

    return str(currentDir.absolute())
